### DISCUSSION TITLE
PiD vs Gyro

### DISCUSSION THREAD

**Discussion by tomt453** [pybricks/support#1848] [General]
Hi All. I'm a coach for an FLL team and we started using Pybricks this year with some pretty good results so far. We are wondering what's the difference between using PiD and the use_gyro function? Does anyone have an example of how to use PiD with a drive base? Many thanks in advance!

**Comment by laurensvalk** [comment]
The `DriveBase` class has the PID controllers already built in. By default, it will use the rotation sensors in the motor as the input. With `use_gyro` it will use the gyro as the information source.

You can also adjust the controller gains like `kp` and `kd` if you like, much like when making your own controller.

Of course you can still build your own if you like to learn about controllers. Then you can use the builtin ones to compare the results.
